### PR TITLE
feat(client): add jest snapshot for settings button and certification…

### LINKED ISSUE #48695: Add jest snapshot following setting button and certification page consistent
**Describe the bug**
This PR #48211 makes the button consistent, we can use snapshot to make sure that these section still consistent and make it easier to review future PRs

Here are the docs about snapshot https://jestjs.io/docs/snapshot-testing

Here is an example file from the codebase, that creates a snapshot https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/settings/certification.test.tsx

### PR DESCRIPTION
**Pull request by nayabatir1**
… page consistency

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #48695 

run jest snapshot
```bash
npm run test:source client/src/components/settings/settings-button.test.js
```
```bash
npm run test:source client/src/components/profile/components/timeline-buttons.test.js
```

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
Didn't find an issue, nicely done. You can run `npx jest -u` to update the snapshot.

it may not run on windows, and that's issue in the library itself. If you run into that, running the command in Gitpod environment will do

**Comment by moT01** [comment]
Let us know when this is ready for review @nayabatir1 👍 Also, do you mind completing the checklist in the PR description template (just put an `x` in the brackets if you've done the thing mentioned).

**Comment by scissorsneedfoodtoo** [comment]
Hi @nayabatir1, thank you for all of your hard work here on these tests.

Before moving forward with these snapshot tests, we may need to discuss them a bit more. I'm not as familiar with snapshot testing as @ahmadabdolsaheb and @ojeytonwilliams probably are, but I noticed a couple of things while going through this PR and looking at output from the failed tests.

First, because dates are rendered on the timeline, timezone differences may cause tests to fail. I believe that's what's happening here: https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/3721578044/jobs/6324704916#step:6:2355

It may be possible to set a timezone in `package.json` before running the tests scripts (`"test": "cross-env TZ=America/New_York run-s create:* build:curriculum build-workers test:*"`), though I'm not sure if we want to go that route.

Next, since dates are being localized based on the client's locale, the Chinese and Spanish test suites will always fail against the English snapshot: https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/3721578044/jobs/6324704756#step:6:2713

It might be possible to have different snapshots for each locale and run tests against those. Or only run snapshot tests against the English client. But again, we probably don't want to go that route.

What might be better is to have more granular tests similar to the ones in `client/src/components/settings/certification.test.tsx`, or even E2E tests in Cypress to check these views and buttons.

What do you all think?

**Comment by nayabatir1** [comment]
@scissorsneedfoodtoo thanks,
I can try to mock dates too.
If you could let me know the file name.

**Comment by ojeytonwilliams** [comment]
@scissorsneedfoodtoo I'm not sure it's the best approach, but setting `TZ` was how I handled this for Chapter.  I added a `global-setup` file

```
module.exports = async () => {
  process.env.TZ = 'UTC';
};
```

and used that in the jest config as

```
  globalSetup: './global-setup.js',
```

Here's the PR https://github.com/freeCodeCamp/chapter/pull/631

**Comment by raisedadead** [comment]
We are officially on Node.js 18 now. We have changed the merge requirements accordingly.

You will need to rebase this PR against the main branch OR Click the "Update branch" button to get the new checks going.

Thanks for your contribution.

**Comment by nayabatir1** [comment]
Hey,
I'm mocking `Date.prototype.toLocaleString` and `Date.prototype.toISOString`. So irrespective to time-zone you'll get same output. Hopefully it will solve the ci/cd failed tests

**Comment by nayabatir1** [comment]
Need suggestion.
 [ci/cd issue](https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/3734145330/jobs/6335966260#step:6:2158)

`showUpcomingChanges` value in `env.json` is `false` due to which while contructing link `jsAlgoBase` is used instead of `jsAlgo22Base` 
Shall I update `env.json` and snapshot ?

**Comment by naomi-lgbt** [comment]
I definitely feel like we shouldn't be mocking the Date API.

**Comment by nayabatir1** [comment]
Do let me know if I missed something, as this PR is marked as draft.

**Comment by raisedadead** [comment]
> Do let me know if I missed something, as this PR is marked as draft.

Draft PRs are not ready for review or same as WIP PRs. The idea is to cut the noise while you are not done with it.

In this particular PR's case the last few comments seem to suggest you are not sure of the approach and need more help (can be done here or on discord chat) and the tests arent passing yet.

You can just mark it ready when it is ready for review again (the approach is finalised and the tests are green).

Cheers.

**Review by raisedadead — CHANGES_REQUESTED** [review]
Several things:

1. You need to move the mocks away to someplace else where they are not bundled with the "source" code.
2. Change the files to use `kebab-case` in their naming.
3. I am not sure why we could not have json data for the mocks? We already have such a copy in the seeding scripts.

**Comment by nayabatir1** [comment]
- [x] You need to move the mocks away to someplace else where they are not bundled with the "source" code.
- [x]  Change the files to use `kebab-case` in their naming.
- [x] I am not sure why we could not have json data for the mocks? We already have such a copy in the seeding scripts.

**Comment by raisedadead** [comment]
Sorry, I should have spelled this out. I think we have a mocks location at `client/src/__mocks__`, we should use that.

File: tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "tools/challenge-auditor/index.ts",
     "tools/challenge-editor/**/*",
     "tools/challenge-helper-scripts/**/*.ts",
-    "tools/scripts/**/*.ts"
+    "tools/scripts/**/*.ts",
+    "client/src/__tests__/settings-button.test.js",
+    "client/src/__tests__/timeline-buttons.test.js"
```
Comment (raisedadead) [inline tsconfig.json:13]: I am almost sure there is a better way to address these config needs; @ojeytonwilliams what do you think?

**Review by raisedadead — CHANGES_REQUESTED** [review]
`client/src/components/profile/components/completedChallenges.js` Use kebab-case.

File: tsconfig.json (hunk 1 above)
Comment (nayabatir1) [reply tsconfig.json:13]: Just to let you know these are auto generated.

Comment (raisedadead) [reply tsconfig.json:13]: Elaborate the steps?

Comment (nayabatir1) [reply tsconfig.json:13]: Basically I ran these command in cli it was auto generated 

`npm run test:source client/src/__tests__/settings-button.test.js`
`npm run test:source client/src/__tests__/timeline-buttons.test.js`

Comment (raisedadead) [reply tsconfig.json:13]: Those commands should not add any changes to the tsconfig.

We usually keep the test spec alongside the code component we are testing. You have chosen to keep it in `__tests__`, which is fine.

Why you're choosing not to use TypeScript here?

**Comment by nayabatir1** [comment]
I used this [jest gatsby](https://www.gatsbyjs.com/docs/how-to/testing/unit-testing/#using-typescript) for all the necessary configuration.

**Comment by nayabatir1** [comment]
I'm marking this PR as draft as I'll be making a lot of pull request.
Bear with me while I try to fix failed test cases.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**